### ISSUE TITLE
TRSS node offline on Jenkins due to low disk space

### ISSUE THREAD

**Issue by llxia** [adoptium/infrastructure#2781]
https://ci.adoptopenjdk.net/computer/trss-node/
<img width="1294" alt="image" src="https://user-images.githubusercontent.com/19273206/185273814-fc5641b0-1577-44a5-bdb6-fa622b25ec04.png">

```
root@trss:~# df -h
Filesystem      Size  Used Avail Use% Mounted on
udev            7.6G  302M  7.3G   4% /dev
tmpfs           1.6G  848K  1.6G   1% /run
/dev/nvme1n1p1  7.7G  7.5G  202M  98% /
tmpfs           7.6G   32M  7.6G   1% /dev/shm
tmpfs           5.0M     0  5.0M   0% /run/lock
tmpfs           7.6G     0  7.6G   0% /sys/fs/cgroup
/dev/nvme0n1p2  167G  8.3G  150G   6% /data
/dev/loop1       27M   27M     0 100% /snap/amazon-ssm-agent/5163
tmpfs           1.6G     0  1.6G   0% /run/user/0
/dev/loop2       56M   56M     0 100% /snap/core18/2409
/dev/loop5       26M   26M     0 100% /snap/amazon-ssm-agent/5656
/dev/loop3      114M  114M     0 100% /snap/core/13308
tmpfs           1.6G     0  1.6G   0% /run/user/1001
/dev/loop0      114M  114M     0 100% /snap/core/13425
/dev/loop4       56M   56M     0 100% /snap/core18/2538
```

**Comment by sxa** [comment]
Can we implement an alternate location for the cache as per the previous comment? We have `/data/jenkins` which could be used for this. Alternatively we could just relocate the jenkins homedir but we need input from those managing the applications running out of there to understand if they will be impacted by such a change.

**Comment by llxia** [comment]
I would prefer to get a brand new machine. The current machine is configured with a limited root dir (7G). We have been running into this problem many times. It is not worth our time and energy to fight for the limited space. Let's pick the future-proof solution and create a new machine with 32GB of root space.

**Comment by sxa** [comment]
It is bad practice to store home direcories and user data on `/` as it can cause security issues with logs not being written properly when user data expands to fill it. All user data should be elsewhere and that would be the case on this system or any other.

It does not make sense to be storing data, which could expand indefinitely, on the root file system. There is 167Gb of space in the /data file system which was provisioned for TRSS' use. Let's use it.

Can you confirm whether moving the home directory would cause a problem for the applications running on there?

**Comment by sxa** [comment]
Regardless of how we resolve this, please provide the details of the capacity requirements of a TRSS server for planning purposes.

**Comment by sxa** [comment]
In the absence of any follow up I have moved the directory to `/data/jenkinshome` and created a symlink from the original location. After this I have restarted the `TRSSFrontend` and `TRSSBackend` services. I do not wish to consider the option of provisioning a new server for this with less than a week to go before a release cycle.

**Comment by smlambert** [comment]
As far as I am aware, it is not TRSS server code or DB data that is the issue now. That was moved under /data a while back. It is the root dir too small as all system apps, logs and /tmp are under root dir. There is not much free space left under root ( only ~700M available last time it was cleaned up).

On another note There seems to be a not so healthy discourse on this topic and I am not sure why.

**Comment by sxa** [comment]
> As far as I am aware, it is not TRSS server code or DB data that is the issue now. That was moved under /data a while back.

Correct - therefore nothing for TRSS should have a reliance on the status of the root file system. I am unclear why we suffered this outage as a result of `/` being full and it would be good to understand that. 

I need someone familiar with the TRSS application to provide input to understand the root cause here since I would not expect the application to to be writing to anything outside the large `/data` filesystem on the machine in a way that would cause an outage such as this.

**Comment by llxia** [comment]
According to Ubuntu wiki, the Root partition: `Size: minimum is 8 GB. It is recommended to make it at least 15 GB. Warning: your system will be blocked if the root partition is full.`

Currently, we have 7.7G total and ~700M free space. And in our case, `/tmp` is within the Root partition. I think asking for a new machine is reasonable.

https://help.ubuntu.com/community/DiskSpace

**Comment by smlambert** [comment]
Reopening as I believe we still need a solution that will work for the public TRSS instance.  I am currently blocked at being able to update the server (via the TRSS_Code_Sync job) because the trss-node can not be added to Jenkins...
Disk space is too low. Only 0.237GB left on /tmp.


Answering a previous question in this issue about usage/requirements, here is a bit of information `df -h` from a different (but similarly loaded) TRSS instance:
```
Filesystem                   Size  Used Avail Use% Mounted on
udev                         7.8G     0  7.8G   0% /dev
tmpfs                        1.6G  1.1M  1.6G   1% /run
/dev/mapper/ubuntu--vg-root  234G   43G  192G  19% /
tmpfs                        7.9G     0  7.9G   0% /dev/shm
tmpfs                        5.0M     0  5.0M   0% /run/lock
tmpfs                        7.9G     0  7.9G   0% /sys/fs/cgroup
/dev/loop1                    56M   56M     0 100% /snap/core18/2566
/dev/loop2                    64M   64M     0 100% /snap/core20/1623
/dev/loop0                    56M   56M     0 100% /snap/core18/2560
/dev/loop3                    48M   48M     0 100% /snap/snapd/16778
/dev/loop4                   109M  109M     0 100% /snap/lxd/23537
/dev/loop5                   136M  136M     0 100% /snap/lxd/23680
/dev/loop6                    62M   62M     0 100% /snap/core20/1611
/dev/loop7                    48M   48M     0 100% /snap/snapd/17029
tmpfs                        1.6G     0  1.6G   0% /run/user/1001
tmpfs                        1.6G     0  1.6G   0% /run/user/0
```

**Comment by sxa** [comment]
Back online and the queued up [TRSS_Code_Sync](https://ci.adoptopenjdk.net/job/TRSS_Code_Sync) jobs have executed successfully. I removed the `.npm` and `.cache` directories from the root user (backed up into `.xz`files in `/data` but I suspect they will never be required.

I've also mounted `/tmp` as a 2GB `tmpfs` (allocated from RAM) file system. At present this won't persist over a reboot, but since jenkins is checking this for space it shoudl protect us from a recurrance of this for now.
